### PR TITLE
Define crypto.Verifier and implement EcdsaSha256Verifier

### PR DESCRIPTION
**Pull request by supl**
This PR defines a Verifier interface in the crypto package and provides an implementation of ECDSA-SHA256.
It also completes godoc for the signer.

### CODE REVIEW

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

File: crypto/verifier.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package crypto
+
+// Verifier defines an interface to verify if the message and the signature are matched.
+type Varifier interface {
```
Comment (thongdk8) [inline crypto/verifier.go]:
```suggestion
type Verifier interface {
```

**Review by thongdk8 — APPROVED** [review]
LGTM except for a typo I marked, please fix it before merging! Thank you!

File: crypto/verifier.go (hunk 1 above)
Comment (supl) [reply crypto/verifier.go]: @thongdk8 
Thank you.
Let me fix this.

Comment (supl) [reply crypto/verifier.go]: Fixes in https://github.com/scalar-labs/scalardl-go-client-sdk/pull/6/commits/87e692b4ea5483c7a77067d7e50ed199632f0d1c

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**